### PR TITLE
[processing] add area report to raster layer unique values algorithm

### PR DESCRIPTION
**Pull request by nirvn**
## Description
This PR adds area alongside pixel count to the raster unique values ~~count~~report algorithm.

As indicated above, I've renamed the algorithm since it now does more than counting pixels.

## Checklist

> Reviewing is a process done by project maintainers, mostly on a volunteer basis. We try to keep the overhead as small as possible and appreciate if you help us to do so by completing the following items. Feel free to ask in a comment if you have troubles with any of them.

- [x] Commit messages are descriptive and explain the rationale for changes
- [x] Commits which fix bugs include `fixes #11111` in the commit message next to the description
- [x] Commits which add new features are tagged with `[FEATURE]` in the commit message
- [x] Commits which change the UI or existing user workflows are tagged with `[needs-docs]` in the commit message and containt sufficient information in the commit message to be documented
- [x] I have read the [QGIS Coding Standards](https://docs.qgis.org/testing/en/docs/developers_guide/codingstandards.html) and this PR complies with them
- [x] This PR passes all existing unit tests (test results will be reported by travis-ci after opening this PR)
- [x] New unit tests have been added for core changes
- [x] I have run [the `scripts/prepare-commit.sh` script](https://github.com/qgis/QGIS/blob/master/.github/CONTRIBUTE.md#contributing-to-qgis) before each commit

### CODE REVIEW

File: python/plugins/processing/tests/testdata/expected/raster_unique_values_report.html [hunk 1]
```diff
@@ -0,0 +1,29 @@
+<html><head><meta http-equiv="Content-Type" content="text/html;charset=utf-8"/></head><body>
+<p>Analyzed file: /home/webmaster/dev/cpp/QGIS/python/plugins/processing/tests/testdata/raster.tif (band 1)</p>
+<p>Extent: 270736.0673250681720674,4458888.9563983473926783 : 270899.8544675338780507,4459029.5745217483490705</p>
```
Comment (nyalldawson) [inline python/plugins/processing/tests/testdata/expected/raster_unique_values_report.html]: It'd be handy to export all these extra properties as outputs from the alg too, so they could be used in later model steps

**Review by nyalldawson — COMMENTED** [review]
Would "Raster Layer Statistics" be a better name here?

**Comment by nirvn** [comment]
@nyalldawson , "raster layer statistics" is already taken by an algorithm that provides basic stats for a given raster layer band.

Using report is a nod to GRASS' r.report module, which provides unique value count & area values too.

**Comment by nirvn** [comment]
@nyalldawson , however, I should probably add statistic as a keyword for this alg.

File: python/plugins/processing/tests/testdata/expected/raster_unique_values_report.html (hunk 1 above)
Comment (nirvn) [reply python/plugins/processing/tests/testdata/expected/raster_unique_values_report.html]: @nyalldawson , you tink so? Which one(s) would you like to be exposed? TOTAL_PIXEL_COUNT, and NODATA_COUNT?

Comment (nyalldawson) [reply python/plugins/processing/tests/testdata/expected/raster_unique_values_report.html]: Expose ALL the things!

Extent, projection (as authid), all the numeric values

They could all be potentially reused later in a model

**Comment by nirvn** [comment]
@nyalldawson , comments addressed in the updated commit. Thanks.

**Comment by nyalldawson** [comment]
Looks good!

**Pull request merged**